model-optimizer/extensions/front/onnx/loop_ext.py

```diff
@@ -57,38 +62,104 @@ def extract(cls, loop_node):
+                                # otherwise create new Parameter
+                                if transit_parameter is None:
+                                    # create new Parameter in inner graph in hope that we will find node later
+                                    param_id = str(inp).split(':')[0]
+                                    cur_graph.add_node(param_id, kind='op', op='Parameter', name=param_id, pb=None,
+                                                       shape=None)
+                                    parameter_node = Node(cur_graph, param_id)
+                                    # need to manually update necessary attrs for the node because extractor will not be called
+                                    # for it because the node does not have .pb attribute
+                                    Parameter.update_node_stat(parameter_node, {})
+                                else:
+                                    parameter_node = transit_parameter
+                                    param_id = transit_parameter.id
+
+                                # create transit parameter in outer graph in hope that real input will be found later
+                                parent_param_id = str(inp).split(':')[0]+"_transit"
+                                parent_graph.add_node(parent_param_id, kind='op', op='Parameter', name=parent_param_id,
+                                                      pb=None, shape=None)
+                                parent_parameter_node = Node(parent_graph, parent_param_id)
+                                # need to manually update necessary attrs for the node because extractor will not be called
+                                # for it because the node does not have .pb attribute
+                                Parameter.update_node_stat(parent_parameter_node, {})
+
+                                external_edges[counter].append(((parent_param_id, 0), parameter_node, parent_param_id))
+                                src_id, src_port = param_id, 0
+                                additional_params[counter][parent_param_id+":0"] = parameter_node
+                                transit_parameter = parent_parameter_node
+
+                            if cur_graph.has_node(id):
+                                assert (cur_graph.has_node(src_id))
+                                edge_attrs = {
+                                    'out': src_port,
+                                    'in': dst_port,
+                                    'name': inp,
+                                    'fw_tensor_debug_info': [(inp, inp)],
+                                    'in_attrs': ['in', 'name'],
+                                    'out_attrs': ['out', 'name'],
+                                    'data_attrs': ['fw_tensor_debug_info']
+                                }
+                                cur_graph.add_edge(src_id, id, **edge_attrs)
```

Comment:
I think it makes sense to create a function to add an edge since this piece of code met twice here. we can eliminate some code.